### PR TITLE
nova quota migration

### PR DESCRIPTION
**Pull request by toha10**
Nova quotas migration fucntionality implemented by Nova API instead
of interraction with DB.

### CODE REVIEW

**Comment by Wasapl** [comment]
Merged build finished. Test PASSed.

**Comment by Wasapl** [comment]
Merged build finished. Test PASSed.

File: cloudferrylib/os/compute/nova_compute.py [hunk 1]
```diff
@@ -57,26 +58,29 @@ def get_client(self, params=None):
...
-                           'resource': quota[1],
-                           'hard_limit': quota[2]},
-                 'meta': {}})
+    def _read_info_quotas(self):
+        service_tenant_id = \
+            self.identity.get_tenant_id_by_name(self.config.cloud.service_tenant)
+        tenant_ids = [tenant.id for tenant in self.identity.get_tenants_list()
+                      if tenant.id != service_tenant_id]
+        user_ids = [user.id for user in self.identity.get_users_list()
+                    if not self.identity.roles_for_user(user.id, service_tenant_id)]
```
Comment (roman-verchikov) [inline cloudferrylib/os/compute/nova_compute.py]: Retrieving the list of all users for real production environment may take hours if keystone uses corporate LDAP as a backend. Let's avoid making this call.

File: cloudferrylib/os/compute/nova_compute.py [hunk 2]
```diff
@@ -403,6 +370,23 @@ def _deploy_flavors(self, flavors, tenant_map):
                     self.add_flavor_access(_flavor['meta']['id'],
                                            tenant_map[tenant])
 
+    def _deploy_quotas(self, quotas, tenant_map, user_map=None):
+        for _quota in quotas:
+            old_tenant_id = _quota['tenant_id']
```
Comment (roman-verchikov) [inline cloudferrylib/os/compute/nova_compute.py]: Does this handle the case when old tenant ID does not exist anymore? E.g. in the following scenario:

1. Tenant `t1` is created
2. Quota `q1` is associated with a tenant `t1`
3. Tenant `t1` is removed
4. Quota `q1` is still associated with non-existing tenant `t1`, will `_quota['tenant_id']` have any tenant ID? Will it fail with `KeyError`?

**Comment by roman-verchikov** [comment]
Please put reasoning behind implementing this in pull request description: what was wrong with direct SQL implementation, and why do we need to rewrite it.

**Comment by roman-verchikov** [comment]
This needs to be rebased

**Comment by roman-verchikov** [comment]
@toha10 please rebase

**Comment by plomakin** [comment]
Failed for me:
```
INFO     [/home/plomakin/jenkins_slave/workspace/cloudferry-functional-tests/cloudferry/cloudferrylib/scheduler/scheduler.py : lineno 47]  05-26 15:21:39  -------- Start task: BaseTask|TransportComputeResources
ERROR    [/home/plomakin/jenkins_slave/workspace/cloudferry-functional-tests/cloudferry/cloudferrylib/scheduler/scheduler.py : lineno 58]  05-26 15:22:05  BaseTask|TransportComputeResources TASK FAILED
Traceback (most recent call last):
  File "/home/plomakin/jenkins_slave/workspace/cloudferry-functional-tests/cloudferry/cloudferrylib/scheduler/scheduler.py", line 71, in process_chain
    self.run_task(task)
  File "/home/plomakin/jenkins_slave/workspace/cloudferry-functional-tests/cloudferry/cloudferrylib/scheduler/scheduler.py", line 63, in run_task
    self.map_func_task[task](task)
  File "/home/plomakin/jenkins_slave/workspace/cloudferry-functional-tests/cloudferry/cloudferrylib/scheduler/scheduler.py", line 92, in task_run
    task(namespace=self.namespace)
  File "/home/plomakin/jenkins_slave/workspace/cloudferry-functional-tests/cloudferry/cloudferrylib/scheduler/task.py", line 96, in __call__
    result = self.run(**namespace.vars)
  File "/home/plomakin/jenkins_slave/workspace/cloudferry-functional-tests/cloudferry/cloudferrylib/os/actions/transport_compute_resources.py", line 33, in run
    info_res = src_compute.read_info(target=target)
  File "/home/plomakin/jenkins_slave/workspace/cloudferry-functional-tests/cloudferry/cloudferrylib/os/compute/nova_compute.py", line 123, in read_info
    return self._read_info_resources(**kwargs)
  File "/home/plomakin/jenkins_slave/workspace/cloudferry-functional-tests/cloudferry/cloudferrylib/os/compute/nova_compute.py", line 109, in _read_info_resources
    self._read_info_quotas()
  File "/home/plomakin/jenkins_slave/workspace/cloudferry-functional-tests/cloudferry/cloudferrylib/os/compute/nova_compute.py", line 73, in _read_info_quotas
    if not self.identity.roles_for_user(user.id, service_tenant_id)]
  File "/home/plomakin/jenkins_slave/workspace/cloudferry-functional-tests/cloudferry/cloudferrylib/os/identity/keystone.py", line 225, in roles_for_user
    return self.keystone_client.roles.roles_for_user(user_id, tenant_id)
  File "/home/plomakin/jenkins_slave/workspace/cloudferry-functional-tests/cloudferry/cloudferrylib/utils/proxy_client.py", line 78, in __call__
    raise e
HttpNotImplemented: User roles not supported: tenant ID required (HTTP 501)
INFO     [/home/plomakin/jenkins_slave/workspace/cloudferry-functional-tests/cloudferry/cloudferrylib/scheduler/scheduler.py : lineno 76]  05-26 15:22:05  Failed processing CHAIN MIGRATION
```

**Comment by roman-verchikov** [comment]
This was superseded by https://github.com/MirantisWorkloadMobility/CloudFerry/pull/211

**Pull request closed without merging**